webfinger.py

```diff
@@ -58,13 +58,20 @@ def template_vars(self, domain, url=None):
 Couldn't find a <a href="http://microformats.org/wiki/representative-hcard-parsing">\
 representative h-card</a> on %s""" % resp.url)
 
-        acct = '%s@%s' % (domain, domain)
-        logging.info('Generating WebFinger data for %s', acct)
+        logging.info('Generating WebFinger data for %s', domain)
         key = models.MagicKey.get_or_create(domain)
         props = hcard.get('properties', {})
         urls = util.dedupe_urls(props.get('url', []) + [resp.url])
         canonical_url = urls[0]
 
+        acct = '%s@%s' % (domain, domain)
+        for url in urls:
+            if url.startswith('acct:'):
+                urluser, urldomain = util.parse_acct_uri(url)
+                if urldomain == domain:
+                    acct = '%s@%s' % (urluser, domain)
```

Comment:
mind logging here that we're use this overridden `acct:` url?

Reply:
done